ingenialink/ethercat/network.py

```diff
@@ -516,23 +519,27 @@ def _notify_status(self, slave_id: int, status: NET_DEV_EVT) -> None:
         for callback in self.__observers_net_state[slave_id]:
             callback(status)
 
-    def _recover_from_power_cycle(self, status: NET_DEV_EVT) -> None:
-        """Transition the slaves to Pre-Op state after a power cycle.
+    def _recover_from_disconnection(self) -> bool:
+        """Recover the CoE communication after a disconnection.
+        All the connected slaves need to transitioned to the PreOp state.
 
-        Args:
-            status: The network status.
-
-        Raises:
-            ILStateError: If not all slaves reach PreOp state.
+        Returns:
+            True if all the connected slaves reach the PreOp state.
 
         """
         self._ecat_master.read_state()
-        if status == NET_DEV_EVT.ADDED and self._ecat_master.state == pysoem.INIT_STATE:
-            self.__init_nodes()
-            if not self._check_node_state(self.servos, pysoem.PREOP_STATE):
-                raise ILStateError(
-                    "The communication cannot be recovered. Not all slaves reached PreOp state"
-                )
+        if self._ecat_master.state > pysoem.INIT_STATE:
+            return True
+        self.__init_nodes()
```

Comment:
During this process, what happens if another thread requests a SDO Read?
Let's say the display widget update process

Reply:
The drive is disconnected at this point, no SDO operations will be attempted.

Reply:
An exception will be raised.